company-capf.el

```diff
@@ -165,11 +165,14 @@
      (plist-get (nthcdr 4 (company--capf-data)) :company-require-match))
     (`init nil)      ;Don't bother: plenty of other ways to initialize the code.
     (`post-completion
-     (company--capf-post-completion arg))
-    ))
+     (company--capf-post-completion arg))))
```

Comment:
This, with all respect, is a bit bizarre since every lisp function is by definition a list.  But I agree it has no place in this PR, so I removed it.

Reply:
Not sure what you mean, I'm just saying that pure-addition or pure-deletions diff hunks are a tiny bit nicer, when one can manage that. So thanks.

Reply:
To continue this bike-shedding, I am saying that you could extend yourreasoning to _any_ lisp function, and make the argument that closing and opening parenthesis should always exist on a line of their own, because everytime you add or remove a form from a lisp function it's a "pure addition" or a "pure deletion". 

This would be bizarre because that's clearly bad lisp style.  _Why_ is that bad lisp style, you ask? I don't know, it's an axiom for me and presumably many others.

Reply:
You could, but then it would indeed look very ugly, whereas two closing parens on a separate line look okay to me. So I'd only apply this approach to pieces of code where the shape of the future changes is fairly obvious.

I'm not the original author of that code, though. That was Stefan. I just left that bit as-is.

Reply:
BTW, grep the Emacs code base for "^ *)+" sometime.

Reply:
> That was Stefan

It could've been McCarthy himself, I still would find it silly :-) But more to our point: doesn't that mean @monnier would be a good reviewer for this change?

> BTW, grep the Emacs code base for "^ *)+" sometime.

Emacs != Lisp style guide ;-)

Reply:
Bike shedding, Yay!!  I'll agree with both of you, for a change: closing parens on their own line are ugly, I'm not sure why I left them there.  But "extending the reasoning" doesn't really work: while all "sexps" can be considered as lists, Elisp is mostly made of function calls and macro calls and a large fraction of them have a fixed number of arguments, so adding/removing elements from those lists is not going to happen.

IOW I vote for green with yellow stars.

Reply:
Ah hah. All right then. :)

Reply:
@monnier, well not really, because blah blah blah <dissolves into nothingness> 

Anyway, this wasn't the reason I summoned you, but in the meantime have posted the matter to emacs-devel at Dmitry's suggestion, so let's continue there.

Reply:
Nevermind, I see that you have answered below

Reply:
> @monnier, well not really, because blah blah blah

I think this is rather rude: my dog opined she's offended.  But I guess I could agree to green with blah blah blah, as long as the blah blah blah is using a sufficiently subdued color.  [ Sorry, but I'm stuck on a proof.  ]